File: JavaScript/8js 1/drawStairs.js
/*
|--------------------------------------------------------------------------
| Draw stairs
|--------------------------------------------------------------------------
|
*/

function drawStairs(n) {
   res = [];
   for (let i = 0; i < n; i++) {
      res.push(" ".repeat(i) + "I");
   }
   return res.join("\n");
}


const drawStairs = (n) => [...Array(n)].map((_, i) => " ".repeat(i) + "I").join("\n");


function drawStairs(n) {
   return Array(n)
      .fill("I")
      .map((e, i) => e.padStart(i + 1, " "))
      .join("\n");
}


console.log(drawStairs(1));
console.log(drawStairs(3));
console.log(drawStairs(5));

File: Python/8py 1/draw_stairs.py
"""
|--------------------------------------------------------------------------
| Draw stairs
|--------------------------------------------------------------------------
|
"""

def draw_stairs(n):
   return '\n'.join([' '*i + 'I' for i in range(n)])


draw_stairs = lambda n: '\n'.join(["{}I".format(" "*e) for e in range(n)])


def draw_stairs(n):
   return 'I\n'.join(map(' '.__mul__, range(n))) + 'I'


print(draw_stairs(3))